### PR TITLE
Epmlabsbrn-171-c UI Admin Angular: make pages for loading resourse file/files to S3

### PR DESCRIPTION
**Pull request by JoshuaKeys**
[EPMLABSBRN-171](https://jira.epam.com/jira/browse/EPMLABSBRN-171)

make pages for loading resourse file/files to S3

### CODE REVIEW

File: frontend-angular/src/app/admin/ngrx/reducers.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+export interface AdminState {
+
+}
+export const initialState: AdminState = {};
+const reducer = createReducer(
+    initialState,
+)
+export function adminReducer(state: AdminState, action: Action) {
+    return reducer(state, action);
+}
```
Comment (pandamorphism) [inline frontend-angular/src/app/admin/ngrx/reducers.ts]: this code does nothing

File: frontend-angular/src/app/admin/services/folders/folder.service.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        for (const key of arrKeys) {
+            const fileOrFolderName = key.textContent;
+            if (fileOrFolderName.indexOf('.') > 0) {
+                continue;
+            } else {
+                folders.push(key);
+            }
+        }
+        return folders;
+    }
```
Comment (pandamorphism) [inline frontend-angular/src/app/admin/services/folders/folder.service.ts]: this method not used

File: frontend-angular/src/app/admin/services/folders/folder.service.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { of, Observable } from 'rxjs';
+import { delay, tap, mergeMap } from 'rxjs/operators';
+import { map } from 'fp-ts/lib/ReadonlyRecord';
+import {pluck} from 'rxjs/operators';
+
+@Injectable()
+export class FolderService {
+
+    getFolders(): Observable<Array<string>> {
+        return this.httpClient.get<any>(`/api/cloud/folders`).pipe(
```
Comment (pandamorphism) [inline frontend-angular/src/app/admin/services/folders/folder.service.ts]: you can  return `this.httpClient.get<string[](......>` and correct type will be inferred to caller

File: frontend-angular/src/app/admin/services/format/format.service.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from '@angular/common/http';
+
+@Injectable()
+export class FormatService {
+    getFormat(seriesId: number) {
+        return this.httpClient.get('/api/fileFormat/'+seriesId);
+    }
```
Comment (pandamorphism) [inline frontend-angular/src/app/admin/services/format/format.service.ts]: this method not used

File: frontend-angular/src/app/admin/services/upload/upload.service.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    getUploadData(folder: string, filename: string): Observable<GetUploadModel> {
+        const resolvedPath =  folder + filename;
+        return this.httpClient.get<GetUploadModel>(`/api/cloud/upload?filePath=${resolvedPath}`);
+    }
+    sendFormData2(action: string, body: FormData): Observable<{}> | Observable<HttpErrorResponse> {
+        return this.httpClient.post(action, body);
+    }
+    sendFormData(action: string, body: FormData): Observable<{}> {
+        return this.httpClient.post(action, body);
+    }
```
Comment (pandamorphism) [inline frontend-angular/src/app/admin/services/upload/upload.service.ts]: these are equal

File: frontend-angular/src/app/shared/pure.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 export function showHappySnackbar(snackbar: { open: (...args: any[]) => void }, message: string) {
   snackbar.open(message, ' 😊 ', {duration: 2000});
+  console.log(snackbar)
 }
 
 export function showSadSnackbar(snackbar: { open: (...args: any[]) => void }, err: { message: string }) {
```
Comment (pandamorphism) [inline frontend-angular/src/app/shared/pure.ts]: these functions are not used anymore

File: .gitignore [hunk 7]
```diff
@@ -5,6 +5,14 @@ build/
 !**/src/main/**
 !**/src/test/**
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply .gitignore]: Fixing this rn.

File: frontend-angular/src/app/admin/admin-page.component.ts [hunk 8]
```diff
@@ -83,6 +83,7 @@ export const slideInAnimation =
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class AdminPageComponent implements OnInit {
+  opened = false;
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/admin-page.component.ts]: This property is being used in the template.
On line 11 of admin-page.component.html
    <mat-sidenav #sideNav class="admin__sidenav" mode="side" opened>

File: frontend-angular/src/app/admin/components/home/home.component.spec.ts [hunk 9]
```diff
@@ -19,7 +19,7 @@ describe('HomeComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
+  // it('should create', () => {
+  //   expect(component).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/components/home/home.component.spec.ts]: Fixing this now

File: frontend-angular/src/app/admin/components/load-file/load-file.component.ts [hunk 10]
```diff
@@ -1,50 +1,53 @@
...
-    },
-    UploadService
-  ],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class LoadFileComponent implements OnInit {
-
-  constructor(@Self() private uploadFileService: UploadService,
-              private snackbar: MatSnackBar) {
+  folders: Observable<Array<string>>;
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/components/load-file/load-file.component.ts]: Fixing this now.

File: frontend-angular/src/app/admin/components/load-tasks/load-tasks.component.spec.ts [hunk 11]
```diff
@@ -19,7 +19,7 @@ describe('LoadTasksComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
+  // it('should create', () => {
+  //   expect(component).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/components/load-tasks/load-tasks.component.spec.ts]: Fixed

File: frontend-angular/src/app/admin/components/load-tasks/load-tasks.component.ts [hunk 12]
```diff
@@ -1,49 +1,64 @@
...
+  format$: Observable<string>;
   groups$: Observable<Group[]>;
   tasksGroup: FormGroup;
   series$: Observable<Series[]>;
 
   constructor(private adminAPI: AdminService,
               private fb: FormBuilder,
-              @Self() private uploadFileService: UploadService,
-              private snackbar: MatSnackBar) {
+              private uploadFileService2: UploadService2,
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/components/load-tasks/load-tasks.component.ts]: Twas during a tests of replacement of the class. Nothing serious. I just forgot to change the name, that's all. I'll fix it now.

File: frontend-angular/src/app/admin/components/load-tasks/load-tasks.component.ts [hunk 13]
```diff
@@ -1,49 +1,64 @@
...
+              private snackBarService: SnackBarService) {
   }
-
+  onSubmit() {
+    const formData = new FormData();
+    formData.append('taskFile', this.tasksGroup.get('file').value);
+    formData.append('seriesId', this.tasksGroup.get('series').value.id);
+    this.uploadFileService2.sendFormData2('/api/loadTasksFile?seriesId=1', formData).subscribe(returnData => {
+      // TODO: - Check for other type of errors
+      this.snackBarService.showSadSnackbar('Successfully loaded ' + this.tasksGroup.get('file').value.name);
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/components/load-tasks/load-tasks.component.ts]: going to fix this now.

File: frontend-angular/src/app/admin/model/GetUploadModel.ts [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+            & { 'x-amz-signature': string }
+            & { 'x-amz-signature': string }
+            & { 'key': string }
+            & { 'acl': string }
+            & { 'x-amz-meta-uuid': string }
+            & { 'x-amz-server-side-encryption': string }
+            & { 'x-amz-credential': string }
+            & { 'x-amz-algorithm': string }
+            & { 'file': string }
+            & { 'x-amz-date': string }>
```
[the remark this answers is not in the document]
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/model/GetUploadModel.ts]: yes, it's a way of typing the return value of the getUploadData service. It was returned as an array.

In line 8 of the upload.service.ts file, the getUploadData method uses it.

File: frontend-angular/src/app/admin/ngrx/reducers.ts (hunk 1 above)
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/ngrx/reducers.ts]: yeah, i planed to add ngrx state management logic in the future, this is just the first step.

File: frontend-angular/src/app/admin/services/folders/folder.service.ts (hunk 2 above)
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/services/folders/folder.service.ts]: i'll remove it

File: frontend-angular/src/app/admin/services/folders/folder.service.ts (hunk 3 above)
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/services/folders/folder.service.ts]: yeah.

File: frontend-angular/src/app/admin/services/format/format.service.ts (hunk 4 above)
Comment (JoshuaKeys) [reply frontend-angular/src/app/admin/services/format/format.service.ts]: fixing this

**Pull request merged**